src/subscriptions/walletConnect.ts

```diff
@@ -0,0 +1,177 @@
+import { WalletConnectProvider, BufferBlock, Block } from '../provider/types'
+
+export const WalletConnectSubscriptionsSymbol = Symbol('WalletConnectSubscriptions')
+
+export interface WalletConnectSubscriptions {
+    [WalletConnectSubscriptionsSymbol]: true
+    onAccountsChanged(callback: (accounts: string[]) => void, once?: boolean): () => void
+    onNetworkChanged(callback: (networkId: string) => void, once?: boolean): () => void
+    onChainChanged(callback: (chainId: number) => void, once?: boolean): () => void
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onPayload(callback: (payload: any) => void, once?: boolean): () => void
+    onError(callback: (error: Error) => void, once?: boolean): () => void
+    onConnect(callback: () => void, once?: boolean): () => void
+    onBlock(callback: (block: BufferBlock) => void, once?: boolean): () => void
+    onRawBlock(callback: (block: Block) => void, once?: boolean): () => void
+    onLatestBlock(callback: (block: Block) => void, once?: boolean): () => void
+    onSync(
+        callback: ({ oldBlock, newBlock }: { oldBlock: string; newBlock: string }) => void,
+        once?: boolean,
+    ): () => void
+    onStart(callback: () => void, once?: boolean): () => void
+    onStop(callback: () => void, once?: boolean): () => void
+}
+
+const createWalletConnectSubscriptions = (provider: WalletConnectProvider): WalletConnectSubscriptions => {
+    const onAccountsChanged: WalletConnectSubscriptions['onAccountsChanged'] = (cb, once) => {
```

Comment:
Curiosity: why using `WalletConnectSubscriptions['onAccountsChanged']` rather than `WalletConnectSubscriptions.onAccountsChanged` ?

Reply:
TS sees `WalletConnectSubscriptions.onAccountsChanged` as value, not type
Or something, anyway, it's a TS error like this